README.md
<h1 align="center">IAK博客</h1>

<div align="center">
<img src="https://img.shields.io/badge/-HTML5-E34F26?style=for-the-badge&logo=html5&logoColor=white" />
<img src="https://img.shields.io/badge/-CSS3-1572B6?style=for-the-badge&logo=css3&logoColor=white" />
<img src="https://img.shields.io/badge/-TS-0288d1?style=for-the-badge&logo=typescript&logoColor=white" />
<img src="https://img.shields.io/badge/-ASTRO-e36d25?style=for-the-badge&logo=astro&logoColor=white" />
<img src="https://img.shields.io/badge/-VUE3-41b883?style=for-the-badge&logo=vue.js&logoColor=white" />
</div>

<div align="center">
<img src="https://forthebadge.com/images/badges/built-with-love.svg" />
<img src="https://forthebadge.com/images/badges/built-by-hipsters.svg" />
<img src="https://forthebadge.com/images/badges/open-source.svg" />
<img src="https://forthebadge.com/images/badges/made-with-markdown.svg" />
</div>



# 关于
本博客使用[Astro框架](https://astro.build/)开发，组件使用了astro组件和vue组件（vue开发功能比较方便，但本项目是构建静态站点，不是SSR，最终页面呈现时vue组件会有一段时间空白。后期可能会将vue组件慢慢改回astro组件）

示例网站见[Iak示例博客](https://iak.dorakika.cn)、[DORAKIKA的博客](https://blog.dorakika.cn)

![20230407141533.png](https://img.dorakika.cn/md/20230407141533.png)

![20230407141735.png](https://img.dorakika.cn/md/20230407141735.png)

# 特色
- 🎨多种主题配置
- 📄支持文章页、豆瓣书影音标记、友链页、说说页
- 🤖友好的SEO、OpenGraph支持、sitemap与robot.txt
- 🖼️支持生成页面分享图片
- 📊站点面板：标签集合、最近豆瓣标记
- 🔀随机文章跳转
- 🔍简易搜索功能
- 🔔rss订阅支持
- ✨友好的键盘交互

# 如何使用
```shell
# 克隆仓库
git clone ...
# 安装依赖
yarn

# 本地开发
yarn dev

# 本地构建
yarn build

# 本地预览
yarn preview
```

# 部署
目前只做了vercel的（因为我用的就是vercel），从vercel直接导入仓库即可，其他可以到Astro官网自行适配（问就是懒

# 使用
- 主要配置内容在`/src/config.ts`中
- 文章内容在`/src/content/posts/`内，后续可能会添加其他形式的内容（目前设想中会添加一种用来记录某段时间事情的文档模式（差不多是日记周记这个概念））
- 友链内容在`/src/content/friends/_index.yml`内配置
- 书影音游等标记在`/src/content/mark/_index.yml`内配置
- 新建页面参考`src/pages/friends.astro`建立
- 启动[twikoo](https://twikoo.js.org/quick-start.html)评论需先部署并获取服务地址（其他评论服务，emm，我只能说：欢迎PR）

CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

### [0.10.5](///compare/v0.10.4...v0.10.5) (2023-06-04)


### Features

* 添加图片错误后的默认图片替换 762d11a
* 文章底部额外信息组件、文章数据获取逻辑抽离 f014d58
* deploy脚本可配置ssh端口 b7a437b


### Bug Fixes

* 豆瓣同步脚本bug 125f69c
* 文件导入大小写修复 e2e0aea
* 文章头图布局bug 921d1f4
* Grid布局错误修复 0c4d308
* PostSlider排序修复 b965e8f

### [0.10.4](///compare/v0.10.3...v0.10.4) (2023-05-16)


### Features

* 颜色修改 65a177c
* 组件主题色属性修改为color属性 af420cd
* header-bg配置 cd84ff3
* mark时间轴布局 030fa0d

### [0.10.3](///compare/v0.10.2...v0.10.3) (2023-05-11)


### Features

* 搜索组件夜间样式适配 f33d012
* 选中文本颜色样式 e30f5d7
* 窄屏下文章meta信息简化 6bca9df
* 组件夜间模式降低亮度 ed0fed7
* header的bg和bottom可配置 72ad06b
* recent memory 样式修改 2504a55


### Bug Fixes

* 修复tabs组件key重复问题 bb4b3ef

### [0.10.2](///compare/v0.10.1...v0.10.2) (2023-05-04)


### Features

* memory可选配置card样式 8378d2f


### Bug Fixes

* header字体颜色 28f73f9

### [0.10.1](///compare/v0.10.0...v0.10.1) (2023-04-29)


### Features

* 样式调整 5b62d92
* memory页面，样式修改为时间轴 726d90d

## [0.10.0](///compare/v0.9.2...v0.10.0) (2023-04-28)


### Features

* 惰性注册的右键菜单 e94596d
* 样式修改 32ae20e
* 阅读体验优化，窗口resize后恢复至之前阅读位置 92eafd6
* md website 组件 f8ef26f


### Bug Fixes

* 修复tabs组件嵌套失效bug e37e0f3

### [0.9.2](///compare/v0.9.1...v0.9.2) (2023-04-17)


### Features

* 顶栏固定效果优化 ed9020f
* slider样式优化 23d3d47
* twikoo样式优化 2ffb27d


### Bug Fixes

* mark样式及日期格式化 2cacc57
* nav超过header才变色 846c2a6

### [0.9.1](///compare/v0.9.0...v0.9.1) (2023-04-15)


### Features

* 滚动条样式 49c0fb0
* 颜色修改 f20caf8
* copyright样式 38d2f7e
* md写作，友链Friend组件 fd3d608
* toolbar样式 d760d52


### Bug Fixes

* margin 问题 a32be29

## [0.9.0](///compare/v0.8.0...v0.9.0) (2023-04-11)


### Features

* 卡片链接LinkCard组件 912756f
* a链接md组件 4666e4b
* tabs组件-md 854ff55


### Bug Fixes

* 修复由于聚焦按钮导致的滚动中断问题 c273bf0
* 已知样式问题修复 a808fa9

## [0.8.0](///compare/v0.7.3...v0.8.0) (2023-04-07)


### Features

* **content:** 去除Tip组件，转为Outline组件，添加Grid布局组件 60345d6

### [0.7.3](///compare/v0.7.2...v0.7.3) (2023-04-04)


### Features

* 键盘操作体验优化 82729ee
* 添加时间检测 b50f9b4
* **css:** focus样式修改 63039cb


### Bug Fixes

* 404页面头部 92f87aa

### [0.7.2](///compare/v0.7.1...v0.7.2) (2023-04-02)


### Features

* 添加多种头部背景效果 d812b10


### Bug Fixes

* 滚动逻辑优化，tab键聚焦时自动恢复布局 4704044
* 仅shift+滚轮会触发滚动 20b78b1
* 夜间模式切换同步面板配置 ed3cf43

### [0.7.1](///compare/v0.7.0...v0.7.1) (2023-04-01)


### Bug Fixes

* 修复误操作带来的样式bug 8b4109f
* 页面头部布局填充 fca4da9

## [0.7.0](///compare/v0.6.0...v0.7.0) (2023-04-01)


### Features

* 图片fancybox添加 97664b1
* 增强tab键交互样式 1a685f6
* mdx代码块实现折叠、全屏、复制功能 618847c


### Bug Fixes

* rss误操作恢复 3aff211

## [0.6.0](///compare/v0.5.3...v0.6.0) (2023-03-24)


### Features

* 添加页面与昨日书、首页添加近期碎片 48cd471
* Markdown样式美化：链接，mark，引用 195f514
* mdx组件自动导入 e37513a

### [0.5.3](///compare/v0.5.2...v0.5.3) (2023-03-18)


### Features

* 简陋的夜间模式、额外的文章卡片和轮播、一些样式修改 35009f5
* css 修改 01733d6


### Bug Fixes

* 修复关闭twikoo时仍旧调用twikoo的问题 3bd457e

### [0.5.2](///compare/v0.5.1...v0.5.2) (2023-03-15)


### Features

* 豆瓣同步脚本更新，mark-item类型改为large、medium、small 071956e
* 首页添加最近评论 4f2cf0b
* 文章页添加字数统计 55afe3d
* 友链头像懒加载 f37b78e
* 站点面板近期标记显示icon f186349


### Bug Fixes

* 修复首页轮播图个数限制失效的问题 f5aef69

### [0.5.1](https://github.com/DORAKIKA/iak/compare/v0.5.0...v0.5.1) (2023-03-14)


### Features

* 标记面板和标签面板合并到站点面板中 ([ced9067](https://github.com/DORAKIKA/iak/commit/ced906754b8703b10820f39136342bd8e1686f06))
* 将主题配置移入全站面板中，并适配样式 ([c49e8d4](https://github.com/DORAKIKA/iak/commit/c49e8d47daf510c72f3ab3372f3ae45dad884ce2))
* 可选白色卡片配置 ([c2b9ca2](https://github.com/DORAKIKA/iak/commit/c2b9ca22c0accc10b0466418f146b39122f81982))
* 样式修改 ([37226f6](https://github.com/DORAKIKA/iak/commit/37226f6b37469ee691f519d46f34d57360d8f4cb))


### Bug Fixes

* 补充说明 ([cbd3a3d](https://github.com/DORAKIKA/iak/commit/cbd3a3da54ca84666c17925e0ad9c8a134011478))

## [0.5.0](https://github.com/DORAKIKA/iak/compare/v0.4.3...v0.5.0) (2023-03-13)


### Features

* 面板从全屏更换为卡片弹窗 ([bcfafae](https://github.com/DORAKIKA/iak/commit/bcfafaec6081a92738730774dcef243e9b131f12))
* 添加从Notion同步豆瓣书影音的脚本 ([3d3bed0](https://github.com/DORAKIKA/iak/commit/3d3bed0a579915c814548aa27ae0f3f7666e33f1))
* mark页可选择添加limit限制个数 ([30b1490](https://github.com/DORAKIKA/iak/commit/30b14902d9b1955877a2032cea2304a42fa5efae))

### [0.4.3](https://github.com/DORAKIKA/iak/compare/v0.4.2...v0.4.3) (2023-03-12)


### Features

* 添加@astro/prefetch ([67fd1a3](https://github.com/DORAKIKA/iak/commit/67fd1a3e27381ca81f47348d4520195c73453833))

### [0.4.2](https://github.com/DORAKIKA/iak/compare/v0.4.1...v0.4.2) (2023-03-11)


### Features

* iak全局对象 ([f1a74c0](https://github.com/DORAKIKA/iak/commit/f1a74c0c3c234ff726537ed4c4fdeccdc859989e))


### Bug Fixes

* 取消twikoo的partytown配置 ([404a9ec](https://github.com/DORAKIKA/iak/commit/404a9ecb96965f69c66be28026d0431e50ab0d2b))
* 修复mark页所有分组会闪现的问题 ([d9cd281](https://github.com/DORAKIKA/iak/commit/d9cd2817eab511c91f0f768c26b3765cf8dbc244))

### [0.4.1](https://github.com/DORAKIKA/iak/compare/v0.4.0...v0.4.1) (2023-03-09)


### Features

* 书影音标记可切换分组 ([7eeb204](https://github.com/DORAKIKA/iak/commit/7eeb2043f2adb0911d7d3d2101e26fece944ab31))
* 样式修改 ([c7b55e8](https://github.com/DORAKIKA/iak/commit/c7b55e826a7e9994861364c0579a317048e16a00))

## [0.4.0](https://github.com/DORAKIKA/iak/compare/v0.3.4...v0.4.0) (2023-03-05)


### Features

* 添加全站面板与标签云 ([e9677e1](https://github.com/DORAKIKA/iak/commit/e9677e1c98828791de9a7f0e7d1bcfd09425d092))
* 添加一个标记书影音游的页面 ([ce03819](https://github.com/DORAKIKA/iak/commit/ce03819c15a6903b5a23aa357b2c808b29be4382))
* 主题添加图片背景控制 ([460471e](https://github.com/DORAKIKA/iak/commit/460471ea4ca60a80bdcddedffb530f515cdc9db0))
* hover样式 ([08617fa](https://github.com/DORAKIKA/iak/commit/08617fa3f05ee3939a3c43b135a68a952bbe1fb9))
* tag页面 ([a692e6b](https://github.com/DORAKIKA/iak/commit/a692e6bb20f3bb502b2a83dfbdf331a45b9b6aa9))


### Bug Fixes


src/config.ts
import type { 
    MenuItem, 
    SiteConfig, 
    DefaultAssets, 
    PostSliderConfig, 
    TwikooConfig, 
    FooterConfig, 
    MemoriesConfig, 
    ThemeConfig, 
    HeadingConfig
} from './types/config'

// 站点信息
export const site:SiteConfig = {
    // 你将要部署到的站点地址
    origin: 'https://iak.dorakika.cn',
    // 站点图标链接
    favicon: '/iak.png',
    // 站点名称
    name: 'IAK',
    // 站点标语
    slogan: '一个简单的Astro博客',
    // 站点描述
    description: '这里是一个基于Astro框架搭建的简单博客，主要使用Astro原生与Vue3.0+TypeScript开发',
    // 站点关键词
    keywords: 'Astro,博客,Vue3.0,TypeScript',
    // 站点作者
    author: 'DORAKIKA'
}
export const author = {
    // 作者名称
    name: 'DORAKIKA',
    // 作者头像
    avatar: 'https://q.qlogo.cn/g?b=qq&nk=1633198089&s=140',
    // 作者简介
    description: '我是一个喜欢折腾的前端工程师，对一切新鲜事物充满好奇，希望我的文章能给你带来思考和帮助',
    // 作者邮箱
    email: 'dorakika@163.cm',
}
// 顶部菜单：仅支持二级菜单
export const menus: MenuItem[] = [
    {
        name: '时',
        children: [
            {name: '文章归档', href: '/posts'},
            {name: '与昨日书', href: '/memories'},
            {name: '吾之足迹', href: '/mark'}
        ]
    },{
        name: '空',
        children: [
            {name: '友情链接', href: '/friends'}
        ]
    }
]

export const default_assets: DefaultAssets = {
    // 默认文章封面
    cover: '/images/placeholder.png'
}

// 页脚信息
export const footer: FooterConfig = {
    beian: {
        enabled: true,
        value: "皖ICP备2022003488号"
    },
    slogan: "这里是一个简简单单的博客示例 - qaq",
    links: [
        {
            href: 'https://dorakika.cn',
            icon: '<img src="/images/dorakika.png" class="icon" alt="DORAKIKA" />',
            title: 'DORAKIKA'
        },
        {
            href: 'https://github.com/dorakika/iak',
            icon: '<svg t="1670852782966" class="icon" viewBox="0 0 1024 1024" version="1.1" xmlns="http://www.w3.org/2000/svg" p-id="2656" width="32" height="32"><path d="M0 524.992q0 166.016 95.488 298.496t247.488 185.504q6.016 0.992 10.016 0.992t6.496-1.504 4-3.008 2.016-4.992 0.512-4.992v-100.512q-36.992 4-66.016-0.512t-45.504-14.016-28.992-23.488-16.992-25.504-8.992-24-5.504-14.496q-8.992-15.008-27.008-27.488t-27.008-20-2.016-14.496q50.016-26.016 112.992 66.016 34.016 51.008 119.008 30.016 10.016-40.992 40-70.016Q293.984 736 237.984 670.976t-56-158.016q0-87.008 55.008-151.008-22.016-64.992 6.016-136.992 28.992-2.016 64.992 11.488t50.496 23.008 25.504 17.504q56.992-16 128.512-16t129.504 16q12.992-8.992 28.992-19.008t48.992-21.504 60.992-9.504q27.008 71.008 7.008 135.008 56 64 56 151.008 0 92.992-56.992 158.496t-172 85.504q43.008 43.008 43.008 104v128.992q0 0.992 0.992 3.008 0 6.016 0.512 8.992t4.512 6.016 12 3.008q152.992-52 250.496-185.504t97.504-300.512q0-104-40.512-199.008t-108.992-163.488-163.488-108.992T512.032 12.96 313.024 53.472 149.536 162.464t-108.992 163.488-40.512 199.008z" p-id="2657"></path></svg>',
            title: 'Github'
        },{
            href: 'mailto:dorakika@163.com',
            icon: '<svg t="1670853960118" class="icon" viewBox="0 0 1024 1024" version="1.1" xmlns="http://www.w3.org/2000/svg" p-id="1440" width="32" height="32"><path d="M874.666667 375.189333V746.666667a64 64 0 0 1-64 64H213.333333a64 64 0 0 1-64-64V375.189333l64 54.250667V746.666667h597.333334V429.44l64-54.250667zM810.666667 213.333333a64.789333 64.789333 0 0 1 22.826666 4.181334 63.616 63.616 0 0 1 26.794667 19.413333 64.32 64.32 0 0 1 9.344 15.466667c2.773333 6.570667 4.48 13.696 4.906667 21.184L874.666667 277.333333v21.333334L553.536 572.586667a64 64 0 0 1-79.893333 2.538666l-3.178667-2.56L149.333333 298.666667v-21.333334a63.786667 63.786667 0 0 1 35.136-57.130666A63.872 63.872 0 0 1 213.333333 213.333333h597.333334z m-9.6 64h-578.133334L512 523.882667 801.066667 277.333333z" p-id="1441"></path></svg>',
            title: 'Mail'
        },{
            href: '/rss.xml',
            icon: '<svg t="1677297152241" class="icon" viewBox="0 0 1024 1024" version="1.1" xmlns="http://www.w3.org/2000/svg" p-id="3739" width="128" height="128"><path d="M677.7 512c0-91.135-74.565-165.7-165.7-165.7s-165.7 74.565-165.7 165.7c0 77.879 53.852 143.33 126.761 160.73-1.657 4.143-2.486 8.285-2.486 13.257v281.692c0 23.198 18.227 41.425 41.425 41.425s41.425-18.227 41.425-41.425v-281.692c0-4.143-0.83-9.114-2.486-13.257 72.909-17.398 126.761-82.851 126.761-160.73zM512 603.135c-50.539 0-91.135-40.597-91.135-91.135s40.597-91.135 91.135-91.135 91.135 40.597 91.135 91.135-40.597 91.135-91.135 91.135z" fill="#3399FF" p-id="3740"></path><path d="M512 14.897c-274.235 0-497.103 222.867-497.103 497.103 0 166.53 82.022 314.004 207.126 404.309l48.054-67.108c-104.391-75.395-172.329-198.842-172.329-337.201 0-228.667 185.584-414.253 414.253-414.253s414.253 185.584 414.253 414.253c0 138.359-67.938 261.808-173.157 336.373l48.054 67.108c125.933-90.307 207.126-237.78 207.126-404.309 0.83-273.407-222.039-496.274-496.274-496.274z" fill="#3399FF" p-id="3741"></path><path d="M512 180.598c-183.099 0-331.402 148.302-331.402 331.402 0 111.019 54.682 209.612 138.359 269.264l48.054-67.108c-62.966-45.568-103.563-119.306-103.563-202.156 0-137.533 111.019-248.551 248.551-248.551s248.551 111.019 248.551 248.551c0 82.851-40.597 156.588-103.563 202.156l48.054 67.108c83.679-59.652 138.359-158.244 138.359-269.264 0-183.099-148.302-331.402-331.402-331.402z" p-id="3742"></path></svg>',
            title: 'rss订阅'
        }
    ]
}

export const post_slider: PostSliderConfig = {
    limit: 4
}

export const memories: MemoriesConfig = {
    // 在memories页面显示的最大数量
    show_limit: 10,
    // 在首页显示的最大数量
    recent_limit: 5
}

// twikoo => https://twikoo.js.org/
export const twikoo: TwikooConfig = {
    enabled: false,
    url: "https://twikoo.you.love",
    recent_limit: 5,
    // 你的留言板页面，向用户呈现的时候引导用户去留言
    board_url: ''
}

export const headings: HeadingConfig = {
    // 文章标题目录深度（1-6）
    depth: 3
}

// 主题配置
// 此配置只做选项与默认项，具体配置会读取用户浏览器的LocalStorage数据
export const theme: ThemeConfig = {
    main_colors: [
        'var(--blue-meta)',
        'var(--yellow-meta)', 
        'var(--pink-meta)', 
        'var(--red-meta)',
        'var(--green-meta)'
    ],
    large_border_radius: {
        default: true
    },
    brightness: {
        default: 1,
        min: 0.5,
        max: 1,
        step: 0.1
    },
    headerFixed: {
        default: false
    },
    whiteCard: {
        default: true,
    }
}